docs/book/src/component-config-tutorial/testdata/project/Makefile

```diff
@@ -124,9 +124,9 @@ $(KUSTOMIZE): $(LOCALBIN)
 .PHONY: controller-gen
 controller-gen: $(CONTROLLER_GEN) ## Download controller-gen locally if necessary.
 $(CONTROLLER_GEN): $(LOCALBIN)
-	GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION)
+	test -s $(LOCALBIN)/controller-gen || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION)
```

Comment:
Hi, I just came across this PR while fixing a similar issue in our operator's Makefile. I was wondering if these test commands should use the variables for each binary, such as: `test -s $(CONTROLLER_GEN)`. If the user overrides the tool variable (e.g. `CONTROLLER_GEN`), then this test will not work.

Reply:
I think you are right, it should probably be changed to
```Makefile
$(CONTROLLER_GEN): $(LOCALBIN)
	test -s $@ || GOBIN=$(LOCALBIN) go install sigs.k8s.io/controller-tools/cmd/controller-gen@$(CONTROLLER_TOOLS_VERSION)

```